python/pycrdt/text.py

```diff
@@ -103,19 +103,30 @@ def __setitem__(self, key: int | slice, value: str) -> None:
                 del self[key]
                 self.integrated.insert(txn._txn, key, value)
             elif isinstance(key, slice):
-                start, stop = self._check_slice(key)
-                if start != stop:
-                    self.integrated.remove_range(txn._txn, start, stop)
+                start, length = self._check_slice(key)
+                if length > 0:
+                    self.integrated.remove_range(txn._txn, start, length)
                 self.integrated.insert(txn._txn, start, value)
             else:
                 raise RuntimeError(f"Index not supported: {key}")
 
+    def replace_range(self, value: str, start: int, stop: int) -> None:
+        """Replace the range of characters in the interval ['start', 'stop') with 'value'."""
+        del self[start:stop]
+        self[start:start] = value
+
+    def remove_range(self, start: int, stop: int) -> None:
```

Comment:
This method doesn't exist on a builtin Python `list`, and it does the same as:
```py
del text[start:stop]
```
Do you think it is worth it?

Reply:
I would think so. I like to have `insert` and `clear` as verbose option since they carry the reading flow better than brackets and colons, in my opinion.
Another benefit would be that a dedicated method pops up in auto-completion, which might come in handy for new users so they know about this without the need to read a documentation.

Reply:
But `insert` and `clear` are methods of a `list`, and `remove_range` is not.
I'd like to keep the API as small as possible for now.

Reply:
I'm totally fine with that.

I reverted the changes.